certora/helpers/MerkleTrees.sol

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: GNU AGPLv3
+pragma solidity ^0.8.0;
+
+import "./MerkleTreeLib.sol";
+
+contract MerkleTrees {
+    using MerkleTreeLib for MerkleTreeLib.Node;
+    using MerkleTreeLib for MerkleTreeLib.Tree;
+
+    mapping(address => MerkleTreeLib.Tree) trees;
+
+    function newLeaf(
+        address treeAddress,
+        address addr,
+        uint256 value
+    ) public {
+        trees[treeAddress].newLeaf(addr, value);
+    }
+
+    function newInternalNode(
+        address treeAddress,
+        address parent,
+        address left,
+        address right
+    ) public {
+        trees[treeAddress].newInternalNode(parent, left, right);
+    }
+
+    function setRoot(address treeAddress, address addr) public {
+        trees[treeAddress].setRoot(addr);
+    }
+
+    function getRoot(address treeAddress) public view returns (address) {
+        return trees[treeAddress].getRoot();
+    }
+
+    function getLeft(address treeAddress, address addr) public view returns (address) {
+        return trees[treeAddress].getLeft(addr);
+    }
+
+    function getRight(address treeAddress, address addr) public view returns (address) {
+        return trees[treeAddress].getRight(addr);
+    }
+
+    function getValue(address treeAddress, address addr) public view returns (uint256) {
+        return trees[treeAddress].getValue(addr);
+    }
+
+    function getHash(address treeAddress, address addr) public view returns (bytes32) {
+        return trees[treeAddress].getHash(addr);
+    }
+
+    function isEmpty(address treeAddress, address addr) public view returns (bool) {
+        return trees[treeAddress].nodes[addr].isEmpty();
+    }
+
+    function isWellFormed(address treeAddress, address addr) public view returns (bool) {
+        return trees[treeAddress].isWellFormed(addr);
+    }
```

Comment:
Ok so I kept the lib, because I felt like it did a nice separation between working on a tree and working on a mapping of trees.

But you were right there there was some duplication for no reason. I removed the getters from the lib, and it made me realize that the checker is based on the lib (it was importing `MerkleTreeLib` and not `MerkleTrees`). This is not ideal because the verification is based on the contract, so we could potentially have called functions from the lib that were not in the contract. So now everything is based on the contract :+1: